### PR TITLE
Fixed #435 One shot replicator sends STOPPING state immediately

### LINKED ISSUE #435: One shot replicator sends STOPPING state immediately after started replication
- One shot Push Replication sends STOPPING state immediately after started replication.
- One shot Pull Replication sends STOPPING state immediately after ChangeTracker is completed.

Both cases, a replicator is still uploading/downloading documents to/from Sync Gateway or CoudDB.

Expected behavior: STOPPING state should be sent after all replications are completed.

### PR DESCRIPTION
**Pull request by hideki**
- In case of One Shot replication,  STOP_GRACEFUL is fired almost immediately. It makes replicator state STOPPING instead of RUNNING during replication.
- Like continuous replication mode, use waitForPendingStates() to wait till all tasks are completed, then fire STOP_GRACEFULLY at end of waitForPendingState() if mode is OneShot

### CODE REVIEW

File: src/main/java/com/couchbase/lite/replicator/PullerInternal.java [hunk 1]
```diff
@@ -802,7 +802,8 @@ private void processChangeTrackerStopped(ChangeTracker tracker) {
                 if (tracker.getLastError() != null) {
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/PullerInternal.java]: Log above this line doesn't valid anymore.

File: src/main/java/com/couchbase/lite/replicator/PullerInternal.java [hunk 2]
```diff
@@ -940,7 +942,14 @@ public void waitForPendingFutures() {
             Log.e(Log.TAG_SYNC, "Exception waiting for pending futures: %s", e);
         }
 
-        fireTrigger(ReplicationTrigger.WAITING_FOR_CHANGES);
+        // continuous mode, make state IDLE
+        if(isContinuous()) {
+            fireTrigger(ReplicationTrigger.WAITING_FOR_CHANGES);
+        }
+        // one shot mode, make state STOPPING
+        else{
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/PullerInternal.java]: Need space b/w else and {

File: src/main/java/com/couchbase/lite/replicator/PusherInternal.java [hunk 3]
```diff
@@ -178,8 +179,14 @@ public void waitForPendingFutures() {
                 //       This is reason to check if queue is empty.
                 if (pendingFutures.isEmpty()) {
                     Log.v(Log.TAG_SYNC, "[waitForPendingFutures()] state=" + stateMachine.getState());
-                    Log.v(Log.TAG_SYNC, "[waitForPendingFutures()] fireTrigger(ReplicationTrigger.WAITING_FOR_CHANGES);");
-                    fireTrigger(ReplicationTrigger.WAITING_FOR_CHANGES);
+                    if(isContinuous()) {
+                        // Make state IDLE
+                        Log.v(Log.TAG_SYNC, "[waitForPendingFutures()] fireTrigger(ReplicationTrigger.WAITING_FOR_CHANGES);");
+                        fireTrigger(ReplicationTrigger.WAITING_FOR_CHANGES);
+                    }else{
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/PusherInternal.java]: Need space between }, else, and {.

File: src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java [hunk 4]
```diff
@@ -857,7 +857,7 @@ public void onCompletion(HttpResponse httpResponse, Object result, Throwable e)
                     // TODO: with regards to behavior of a continuous replication.
                     // Note: was added in order that unit test testRunReplicationWithError() finished and passed.
                     // (before adding this, the replication would just end up in limbo and never finish)
-                    fireTrigger(ReplicationTrigger.STOP_GRACEFUL);  // TODO: call triggerStop(); just to be more consistent
+                    fireTrigger(ReplicationTrigger.STOP_GRACEFUL);  // TODO: call triggerStopGraceful(); just to be more consistent
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java]: I'm confusing between  fireTrigger(ReplicationTrigger.STOP_GRACEFUL) and triggerStopGraceful(). Does it do the same thing? Why can't we call triggerStopGraceful()?

**Comment by pasin** [comment]
I couldn't follow all logics so I reviewed the code as much as I can do.

File: src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java (hunk 4 above)
Comment (hideki) [inline src/main/java/com/couchbase/lite/replicator/ReplicationInternal.java]: They are same. All method should use `triggerStopGraceful()` for consistency.

**Comment by hideki** [comment]
Applied all feedbacks from @pasin. Please review this PR again.

**Pull request merged**